### PR TITLE
feat: IDP-915-workbook-pagamenti

### PR DESCRIPTION
**Pull request by LarissaASLeite**
### List of changes

Add workbook Payment
Add Alert Onboarding Request

It's necessary to register the secret "alert-slack-idpay" in the key vault.

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [x] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

#### Has This Been Tested?

- [ ] Yes
- [ ] No

### Other information
Target: 
```

```

#### How Has This Been Tested?

- [ ] Unit Test Suite
- [ ] Integration Test Suite
- [ ] Api Tests
- [ ] Load Tests

[Link to test results](https://pagopa.atlassian.net/browse/RTD-XXXX)

---

### CODE REVIEW

File: src/domains/idpay-app/01_keyvault.tf [hunk 1]
```diff
@@ -34,3 +34,11 @@ resource "azurerm_key_vault_secret" "revoked_pi_consumer_connection_uri" {
   name         = "${var.domain}-revoked-pi-consumer-connection-uri"
   value        = "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"$ConnectionString\" password=\"\";"
 }
+
+data "azurerm_key_vault_secret" "alert-slack-idpay" {
+
+  count = var.idpay_alert_enabled.enable == true ? 1 : 0
+
+  name         = "alert-slack-idpay"
```
Comment (antonioT90) [inline src/domains/idpay-app/01_keyvault.tf:42]: let's use more significant name (aligned with already existent "alert-core-notification-slack")
```suggestion
  name         = "alert-idpay-notification-slack"
```

File: src/domains/idpay-app/12_alert.tf [hunk 2]
```diff
@@ -0,0 +1,87 @@
+resource "azurerm_monitor_action_group" "slackIdpay" {
+
+  count = var.idpay_alert_enabled.enable == true ? 1 : 0
+
+  name                = "slackIdpay"
+  resource_group_name = data.azurerm_resource_group.monitor_rg.name
+  short_name          = "send_to_slack"
```
Comment (antonioT90) [inline src/domains/idpay-app/12_alert.tf:7]:
```suggestion
  short_name          = "send_to_idpay_slack"
```

File: src/domains/idpay-app/99_variables.tf [hunk 3]
```diff
@@ -34,6 +34,15 @@ variable "domain" {
   }
 }
 
+variable "idpay_alert_enabled" {
```
Comment (antonioT90) [inline src/domains/idpay-app/99_variables.tf:37]: this structure is a waste, let's just define it as a simple boolean, having default value false
MoreOver, inside uat and prod (see env/* folders) you have to set this variable to true

**Review by antonioT90 — CHANGES_REQUESTED** [review]

File: src/domains/idpay-app/01_keyvault.tf [hunk 4]
```diff
@@ -34,3 +34,11 @@ resource "azurerm_key_vault_secret" "revoked_pi_consumer_connection_uri" {
   name         = "${var.domain}-revoked-pi-consumer-connection-uri"
   value        = "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"$ConnectionString\" password=\"\";"
 }
+
+data "azurerm_key_vault_secret" "alert-slack-idpay" {
+
+  count = var.idpay_alert_enabled.enable == true ? 1 : 0
```
Comment (antonioT90) [inline src/domains/idpay-app/01_keyvault.tf:40]:
```suggestion
  count = var.idpay_alert_enabled.enable ? 1 : 0
```

File: src/domains/idpay-app/12_alert.tf [hunk 5]
```diff
@@ -0,0 +1,87 @@
+resource "azurerm_monitor_action_group" "slackIdpay" {
+
+  count = var.idpay_alert_enabled == true ? 1 : 0
```
Comment (antonioT90) [inline src/domains/idpay-app/12_alert.tf:3]:
```suggestion
  count = var.idpay_alert_enabled? 1 : 0
```

**Review by antonioT90 — CHANGES_REQUESTED** [review]

File: src/domains/idpay-app/12_alert.tf [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    name                    = "slackIdpay"
+    email_address           = data.azurerm_key_vault_secret.alert-slack-idpay[count.index].value
+    use_common_alert_schema = true
+  }
+
+}
+
+resource "azurerm_monitor_scheduled_query_rules_alert_v2" "CompleteOnboarding" {
+
+  count = var.env_short == "p" ? 1 : 0
```
Comment (antonioT90) [inline src/domains/idpay-app/12_alert.tf:19]: change this in order to check enable variable

**Review by antonioT90 — CHANGES_REQUESTED** [review]

File: src/domains/idpay-app/12_alert.tf [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+      targetResourceTypes = [
+        "microsoft.operationalinsights/workspaces"
+      ]
+      windowSize = "P1D"
+    }
+  })
+}
+
+# alert rule set with azapi
+resource "azapi_resource" "CosmosServerSideRetry" {
```
Comment (antonioT90) [inline src/domains/idpay-app/12_alert.tf:127]:
```suggestion
resource "azapi_resource" "alert_CosmosServerSideRetry" {
```

File: src/domains/idpay-app/env/uat/terraform.tfvars [hunk 8]
```diff
@@ -37,6 +37,7 @@ reverse_proxy_be_io            = "10.1.0.250"
 #
 dns_zone_internal_prefix = "internal.uat.cstar"
 external_domain          = "pagopa.it"
+idpay_alert_enabled = true
```
Comment (antonioT90) [inline src/domains/idpay-app/env/uat/terraform.tfvars:40]: this is repeated twice in this file

**Review by antonioT90 — CHANGES_REQUESTED** [review]

**Review by GiovanniMancini — DISMISSED** [review]

**Pull request merged**